static/src/js/containers/MapContainer/MapContainer.tsx

```diff
@@ -438,7 +441,12 @@ export const MapContainer: React.FC<MapContainerProps> = (props) => {
   if (granuleIds.length > 0) {
     granuleIds.forEach((granuleId) => {
       const { collectionId, index } = nonExcludedGranules[granuleId]
-      const granule = { ...granulesMetadata[granuleId] }
+      const granule = { ...granuleMetadata[granuleId] }
+
+      // If the granule hasn't been fetched yet, return
+      if (isEmpty(granule)) {
```

Comment:
I don't love this check for some reason. I don't have big feelings and the alternatives seem annoying too. This probably isn't a productive comment on second thought.